aws/aws.go

```diff
@@ -454,9 +455,37 @@ func deleteVolumes(ec2Clients map[string]ec2Client, volumes []*types.Disk) []err
 				if ctx.DryRun {
 					log.Infof("[AWS] Dry-run set, volume is not deleted: %s:%s, region: %s", vol.Name, vol.ID, region)
 				} else {
-					log.Infof("[AWS] Delete volume: %s:%s", vol.Name, vol.ID)
-					if _, err := ec2Client.DeleteVolume(&ec2.DeleteVolumeInput{VolumeId: &vol.ID}); err != nil {
-						errChan <- err
+					log.Infof("[AWS] Initiate delete volume: %s:%s", vol.Name, vol.ID)
+					var detachError error
+					if vol.State == types.InUse {
+						log.Infof("[AWS] Volume %s:%s is in-use, trying to detach", vol.Name, vol.ID)
+						if _, detachError = ec2Client.DetachVolume(&ec2.DetachVolumeInput{VolumeId: &vol.ID}); detachError == nil {
+							log.Infof("[AWS] Volume %s:%s detach succeeded, waiting for 'available' state...", vol.Name, vol.ID)
```

Comment:
This should go into a separated function.

Reply:
Good catch, moved out to a separated func.